tests/ops/op_math/test_controlled.py

```diff
@@ -746,7 +776,7 @@ def test_toffoli(self):
         op = Controlled(qml.PauliX("c"), ("a", 2))
         decomp = _decompose_no_control_values(op)
         assert len(decomp) == 1
-        assert qml.equal(decomp[0], qml.Toffoli(("a", 2, "c")))
+        assert equal_if_decomposed(decomp, qml.Toffoli(("a", 2, "c")))
```

Comment:
If `decomp` is no longer `Toffoli` (even though I think it should be), then we should use `qml.equal` with the operation we expect instead.

We no exactly what will come out. We should then test against what we actually expect.  More complicated test code means more bugs and harder debugging.